internal/providers/vsphere/pke/workflow/create_node_activity.go

```diff
@@ -122,7 +122,11 @@ func generateVMConfigs(input CreateNodeActivityInput) (*types.VirtualMachineConf
 		return nil, err
 	}
 
-	vmConfig := types.VirtualMachineConfigSpec{}
+	vmConfig := types.VirtualMachineConfigSpec{
+		NumCPUs:           int32(input.VCPU),
+		NumCoresPerSocket: int32(input.VCPU),
```

Comment:
**Note**: I'm kinda out of the loop, but do we have 1 socket only (whatever it refers to)?

Before starting to look into what these config values mean exactly my intuition said `NumCPUs` is probably the total value and `socket * NumCoresPerSocket` should not be over that, but I could be completely wrong, I'm going to search for these, just wanted to start the discussion in the meantime.

Reply:
I think there's no major drawback of putting all the virtual cores into one virtual socket (and I think the above spec means exactly that), but some strange commercial license checks tend to prefer this layout, and that's why I normally opt for it.

Reply:
Additionally, if you want to have a custom layout, you can always set the vcpu value to zero, and that will fall back to the template's settings.

Reply:
Thanks.